### DISCUSSION TITLE
v3.1.0-beta2: update changelog for 3.1.0-beta2 (#471)

### DISCUSSION THREAD

**Discussion by jhillyerd** [inbucket/inbucket#472] [General]
### Added
- Reject mail by origin domain: `INBUCKET_SMTP_REJECTORIGINDOMAINS` (#375)
  - Wildcard support (#412)
- Version flag for `inbucket` cmd (#385)
- STLS support for POP3 (#384)
- ForceTLS flag for SMTP (#402)
- Lua scripting additions:
  - `logger` API for Lua (#407)
  - `before.message_stored` handler (#417, #418)
- `$` is replaced with `:` in filestore paths, for `D:\...` syntax (#449)
- REST Client `transport` support (#463)

### Fixed
- UI & Storage paths in systemd service file (#393)
- Web UI will redirect from `prefix` to `prefix/` (#397)
- Include inlines when listing attachments (#398)
- Fail Inbucket startup if unable to create storage dir (#448)
- Close directory file handles immediately, fixes Windows locking (#457)

<hr /><em>This discussion was created from the release <a href='https://github.com/inbucket/inbucket/releases/tag/v3.1.0-beta2'>v3.1.0-beta2: update changelog for 3.1.0-beta2 (#471)</a>.</em>

**Comment by jonstewart24** [comment]
With the implementation of `INBUCKET_SMTP_REJECTORIGINDOMAINS`, and the wildcard support, is there a way to reject mail from all domains except listed domains? 

For example if I wanted to only allow mail originating from `example.com,example2.com` and reject all other origin domains, is there a way to achieve that currently? 

Thanks for any guidance!

**Reply by jhillyerd** [reply]
There isn't a way to do that with configuration settings right now.

You could do it with Lua though, by modifying https://book.inbucket.org/lua/before-mail-accepted.html -- there is also some general info about the Lua implementation in https://book.inbucket.org/lua/index.html

**Reply by jonstewart24** [reply]
Thanks! Will check this out!
